rio_tiler/io/base.py

```diff
@@ -69,13 +71,25 @@ def tile_exists(self, tile_x: int, tile_y: int, tile_z: int) -> bool:
             bool: True if the tile intersects the dataset bounds.
 
         """
-        tile = Tile(x=tile_x, y=tile_y, z=tile_z)
-        tile_bounds = self.tms.bounds(tile)
+        tile_bounds = self.tms.xy_bounds(Tile(x=tile_x, y=tile_y, z=tile_z))
+
+        try:
+            dataset_bounds = transform_bounds(
+                self.crs, self.tms.rasterio_crs, *self.bounds, densify_pts=21,
+            )
+        except:  # noqa
+            # HACK: gdal will first throw an error for invalid transformation
+            # but if retried it will then pass.
+            # Note: It might return `+/-inf` values
+            dataset_bounds = transform_bounds(
+                self.crs, self.tms.rasterio_crs, *self.bounds, densify_pts=21,
+            )
```

Comment:
it's safer to check coordinates in **TMS** projection. BUT for some reason GDAL/PROJ can error on a first attempt to translate coordinates which overflow the CRS `area_of_use` (should return inf or -inf), on a second attempt it will pass 🤷‍♂️